File: kdvFuncGen.py
import numpy as np

def func(x):
    y=(np.sin(np.pi*x))
    return y

points = 100
x = np.linspace(0,2,points, endpoint=False)
y = func(x)
f = open("KdVInitData.dat", "w")
f.write("{0}\n".format(points))
f.write("{0}\n".format(2.0/(float(points))))
for i in range(points):
    f.write("{0}\n".format(y[i]))
    f.write("{0}\n".format(x[i]))
f.close
g = open("TestInitData.dat", "w")
for i in range(points):
    g.write("{0} {1}\n".format(x[i],y[i]))
g.close

File: kdv_func_gen.cpp
#include <cmath>
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

#include "initFunc.h"


int main()
{
	int size = 100;
	double start = -1.;
	double end = 1.;
	// Choices are cos, sin, sech, & exp
	// cos: cos((2pi/L)*x)
	// sin: sin((2pi/L)*x)
	// sech: sech^2(x-L/2)
	// exp: exp(-(x-L/2)^2)
	std::string func;
	func = "exp";

	double dx = (end-start)/(double(size));
	double y[size];

	if (func=="cos"){
		Init_Cos(dx, y, size, start, end);
	}
	else if (func=="sin"){
		Init_Sin(dx, y, size, start, end);
	}
	else if (func=="sech"){
		Init_Sech(dx, y, size, start, end);
	}
	else if (func=="exp"){
		Init_Exp(dx, y, size, start, end);
	}
	else{
		printf("No allowed initial function chosen!!\n");
		return 0;
	}	

	ofstream initData;
	initData.open ("KdVInitData.dat");
	initData << size << "\n" << dx << "\n";
	for (int i=0;i<size;i++){
		initData << y[i] << "\n" << i*dx << "\n";
	}
	initData.close();
	
}	
